test/org/zaproxy/zap/extension/pscanrulesAlpha/XAspNetVersionScannerUnitTest.java

```diff
@@ -0,0 +1,102 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.parosproxy.paros.core.scanner.Plugin.AlertThreshold;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscanrulesAlpha.XAspNetVersionScanner;
+
+public class XAspNetVersionScannerUnitTest extends PassiveScannerTest {
+
+    @Override
+    protected XAspNetVersionScanner createScanner() {
+        return new XAspNetVersionScanner();
+    }
+
+    @Test
+    public void shouldRaiseAlertWhenResponseContainsXAspNetVersionHeader() throws HttpMalformedHeaderException {
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+
+        msg.setResponseBody("<html></html>");
+        setResponse(msg, "X-AspNet-Version");
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("1/1.1"));
+    }
+
+    @Test
+    public void shouldRaiseAlertWhenResponseContainsXAspNetMvcVersionHeader() throws HttpMalformedHeaderException {
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+
+        msg.setResponseBody("<html></html>");
+        setResponse(msg, "X-AspNetMvc-Version");
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("1/1.1"));
+    }
+
+    @Test
+    public void shouldNotRaiseAlertWhenResponseDoesNotContainsXAspNetVersionHeader() throws HttpMalformedHeaderException {
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+
+        msg.setResponseBody("<html></html>");
+        setResponse(msg, "dummy");
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldNotRaiseAlertWhenAlertThresholdIsLow() throws HttpMalformedHeaderException {
+
+        HttpMessage msg = new HttpMessage();
+        msg.setRequestHeader("GET http://www.example.com/test/ HTTP/1.1");
+
+        msg.setResponseBody("<html></html>");
+        setResponse(msg, "X-AspNet-Version");
+        rule.setLevel(AlertThreshold.LOW);
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    private void setResponse(HttpMessage msg, String header) throws HttpMalformedHeaderException {
```

Comment:
Worth renaming to `setResponseHeader` (to be more explicit in what is being set).